src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -1387,12 +1392,17 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
   }
 
   function unbatchedUpdates<A>(fn: () => A): A {
+    const previousIsUnbatchingUpdates = isUnbatchingUpdates;
     const previousIsBatchingUpdates = isBatchingUpdates;
+    // This is only true if we're nested inside either batchedUpdates or a
+    // lifecycle method.
+    isUnbatchingUpdates = isBatchingUpdates || isPerformingWork;
```

Comment:
Why is this set if `isPerformingWork` and then the consequence is ignored if `isPerformingWork`?